app/core/api/views.py

```diff
@@ -116,3 +121,36 @@ class RecurringEventViewSet(viewsets.ModelViewSet):
     permission_classes = [IsAuthenticated]
     queryset = RecurringEvent.objects.all()
     serializer_class = RecurringEventSerializer
+
+
+@extend_schema_view(
+    list=extend_schema(description="Return a list of all the sponsor partners"),
+    create=extend_schema(description="Create a new sponsor partner"),
+    retrieve=extend_schema(description="Return the details of a sponsor partner"),
+    destroy=extend_schema(description="Delete a sponsor partner"),
+    update=extend_schema(description="Update a sponsor partner"),
+    partial_update=extend_schema(description="Patch a sponsor partner"),
+)
+class SponsorPartnerViewSet(viewsets.ModelViewSet):
+    permission_classes = [IsAuthenticated]
+    queryset = SponsorPartner.objects.all()
+    serializer_class = SponsorPartnerSerializer
+
+    # The following code can be uncommented and used later, but it's being left out
```

Comment:
Good choice about commenting this out. I think what you are going to want to do is install Django filter and create a FilterSet - see examples in the FilterSet section of this page for what I am talking about: https://riptutorial.com/django-rest-framework/example/26208/filtering-examples--from-simple-to-more-complex-ones